app/build.gradle

```diff
@@ -30,10 +30,20 @@ android {
         versionCode 118
         vectorDrawables.useSupportLibrary = true
     }
+
     buildTypes {
         release {
-            minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+            minifyEnabled true
+            shrinkResources true
+            proguardFiles getDefaultProguardFile("proguard-android.txt"),
+                    "proguard-rules.pro"
+            resValue "string", "pebble_content_provider", "com.getpebble.android.provider"
+        }
+
+        debug {
+            applicationIdSuffix ".debug"
+            versionNameSuffix "-debug"
+            resValue "string", "pebble_content_provider", "com.getpebble.android.debug.provider"
```

Comment:
This change is equivalent to not having any content provider exposed in the debug build, as it's nowhere to be found in pebble official SDK (e.g.: https://github.com/pebble/pebble-android-sdk/blob/master/PebbleKit/PebbleKit/src/main/java/com/getpebble/android/kit/Constants.java ).

Practically this means that debug builds cannot interact with third party pebble apps, and this is clearly not an option.

Reply:
Unfortunately the alternative is a content provider conflict. Which means there's no way to run release and debug on the same device. Those who wish to do so can still switch to building release and installing that on their device.